packages/flutter_tools/lib/src/commands/create_plugin.dart

```diff
@@ -0,0 +1,180 @@
+    assert(platformsToAdd.isNotEmpty);
+
+    final List<String> existingPlatforms = getSupportedPlatformsInPlugin(directory);
+    for (final String existingPlatform in existingPlatforms) {
+      // re-generate files for existing platforms
+      templateContext[existingPlatform] = true;
+    }
+
+    int generatedCount = 0;
+    final String description = argResults.wasParsed('description')
+        ? stringArg('description')
+        : 'A new flutter plugin project.';
+    templateContext['description'] = description;
+    generatedCount += await renderTemplate('plugin', directory, templateContext, overwrite: overwrite);
+
+    if (boolArg('pub')) {
+      await pub.get(
+        context: PubContext.createPlugin,
+        directory: directory.path,
+        offline: boolArg('offline'),
+        generateSyntheticPackage: false,
+      );
+    }
+
+    final FlutterProject project = FlutterProject.fromDirectory(directory);
+    final bool generateAndroid = templateContext['android'] == true;
+    if (generateAndroid) {
+      gradle.updateLocalProperties(
+        project: project, requireAndroidSdk: false);
+    }
+
+    final String projectName = templateContext['projectName'] as String;
+    final String organization = templateContext['organization'] as String;
+    final String androidPluginIdentifier = templateContext['androidIdentifier'] as String;
+    final String exampleProjectName = projectName + '_example';
+    templateContext['projectName'] = exampleProjectName;
+    templateContext['androidIdentifier'] = createAndroidIdentifier(organization, exampleProjectName);
+    templateContext['iosIdentifier'] = createUTIIdentifier(organization, exampleProjectName);
+    templateContext['macosIdentifier'] = createUTIIdentifier(organization, exampleProjectName);
+    templateContext['description'] = 'Demonstrates how to use the $projectName plugin.';
+    templateContext['pluginProjectName'] = projectName;
+    templateContext['androidPluginIdentifier'] = androidPluginIdentifier;
+
+    generatedCount += await generateApp(project.example.directory, templateContext, overwrite: overwrite, pluginExampleApp: true);
+    return generatedCount;
+  }
+
+  void _validateRequestedPlatforms(List<String> platforms) {
+    if (platforms == null || platforms.isEmpty) {
+      throwToolExit(_kNoPlatformsMessage,
+        exitCode: 2);
+    }
+    final List<String> disabledPlatforms = <String>[];
+    if (platforms.contains('web') && !featureFlags.isWebEnabled) {
+      disabledPlatforms.add('web');
+    }
+    if (platforms.contains('linux') && !featureFlags.isLinuxEnabled) {
+      disabledPlatforms.add('linux');
+    }
+    if (platforms.contains('macos') && !featureFlags.isMacOSEnabled) {
+      disabledPlatforms.add('macos');
+    }
+    if (platforms.contains('windows') && !featureFlags.isWindowsEnabled) {
+      disabledPlatforms.add('windows');
+    }
+    if (disabledPlatforms.isEmpty) {
+      return;
+    }
+    throwToolExit('''
```

Comment:
this is confusing - we're default to include web and desktop platforms even though for most users they are not available?